libs/vkd3d-shader/spirv.c

```diff
@@ -5085,6 +5148,10 @@ static void vkd3d_dxbc_compiler_emit_output(struct vkd3d_dxbc_compiler *compiler
 
     compiler->output_info[signature_idx].id = id;
     compiler->output_info[signature_idx].component_type = component_type;
+    compiler->output_info[signature_idx].dst_write_mask = (VKD3DSP_WRITEMASK_0 << output_component_count) - 1;
+
+    if (!builtin)
```

Comment:
Not sure I understand the rationale behind this. Are there cases where write_mask would be smaller than num_components? Would that happen if multiple signature outputs share the same physical IO variable?